modules/ROOT/pages/sap/sap-connector-examples.adoc

```diff
@@ -0,0 +1,154 @@
+
+Create a Mule application that receives a function using the SAP connector's *Function source* operation. The *Function Source* operation passes the export parameter back to SAP.
+
+. Create a new *Mule Project* in Anypoint Studio.
+. Add the SAP connector to your project.
+. Drag the SAP *Function Source* operation to the flow.
+. Create a new *SAP Inbound* global element configuration and fill in the environment values.
+. Add a transform message component to the flow and fill in the details based on the metadata.
++
+Example:
++
+image::sap/sap-function-return-response.png[]
++
+The flow should look like this:
++
+image::sap/sap-flow-response.png[]
++
+In the SAP Studio interface, you should see:
++
+image::sap/sap-result-sap-gui.png[]
+
+=== Send an IDoc to SAP
+
+To create a Mule application to send an IDoc to SAP:
+
+. Create a new *Mule Project* in Anypoint Studio.
+. Create a new HTTP Listener global element configuration and use the default values.
+. Drag the HTTP Request operation onto the canvas and configure the following parameters:
++
+[%header%autowidth.spread]
+|===
+|Parameter |Value
+|Connector Configuration | `HTTP_Listener_Configuration`
+|Path |`/sendIDoc`
+|===
++
+. Create a new SAP Outbound global element configuration and fill with environment values.
+. Click Test to confirm that Mule can connect with the SAP instance. If the connection is successful, click OK to save the configuration. Otherwise, review or correct any invalid parameters and test again.
+. Select the SAP module and add Send IDoc operation next to the HTTP and in the Extension Configuration field select the configuration created in the previous section.
+. Configure the operation with the following values:
++
+[%header%autowidth.spread]
+|===
+|Parameter|Value
+|Display Name |Send IDoc (or any other name you prefer).
+|Connector configuration |`SAP_Outbound` - The name of the global element you created.
+|Key |MATMAS01
+|Content |`#[payload]`
+|===
++
+. Drag a *Transform Message* component before the SAP connector, and click the component to open its properties editor. Once metadata has been retrieved, select the respective fields to populate for the Employee. 
++
+The Transform script should look like:
++
+[source,dataweave,linenums]
+----
+%dw 2.0
+output application/xml
+---
+read('<?xml version="1.0"?>
+<MATMAS01>
+    <IDOC BEGIN="1">
+        <EDI_DC40 SEGMENT="1">
+             <TABNAM>EDI_DC40</TABNAM>
+            <MANDT>800</MANDT>
+        </EDI_DC40>
+    </IDOC>
+</MATMAS01>
+',"application/xml")
+----
++
+. Add a Logger right after the SAP endpoint to see the connector payload in the logs.
+. Save and Run as a Mule Application.
+. From a web browser, test the application by entering an employee's internal Id, first name, and last name in the form of the following query parameters:
++
+[source]
+----
+http://localhost:8081/sendIDoc
+----
++
+Mule conducts the query and adds the employee record.
+
+=== XML to Send an IDoc to SAP
```

Comment:
Why is this a separate subtopic rather than part of the previous, as its example?

Reply:
The text about sending an IDoc to SAP is longer than the text about the container topic of returning values to SAP. What is the actual purpose and focus of this topic set?